applications/FluidDynamicsApplication/custom_utilities/fluid_calculation_utilities.cpp

```diff
@@ -65,6 +65,164 @@ void FluidCalculationUtilities::UpdateValue(
     noalias(rOutput) += rInput;
 }
 
+template<>
+void FluidCalculationUtilities::AssignGradientValue(
+    const double& rInput,
+    const double rShapeFunctionDerivative,
+    const IndexType DirectionIndex,
+    array_1d<double, 3>& rOutput)
+{
+    KRATOS_TRY
+
+    KRATOS_DEBUG_ERROR_IF(DirectionIndex > 2)
```

Comment:
Why only in debug?

Reply:
(same comment below)

Reply:
I was thinking, it is a waste of time to check it in runtime, since these methods will be called within templated elements for dimension. So only at the developer level these errors will be found. Once the developement is finished, there is no way (at least as far as i understand) to raise this error.

Reply:
If this is the case I'd directly remove them.

Reply:
done :)